### PR TITLE
tests/unittest.cpp : added unit tests

### PR DESCRIPTION
**Pull request by hg333**
Added unit tests corresponding to https://github.com/CrowCpp/Crow/pull/212, issue:https://github.com/CrowCpp/Crow/issues/21?.

### CODE REVIEW

**Comment by The-EDev** [comment]
Thanks for your work, but checking the `Content-Type` header is more important IMO

File: include/crow/http_response.h [hunk 1]
```diff
@@ -48,6 +48,11 @@ namespace crow
             headers.emplace(std::move(key), std::move(value));
         }
 
+        std::string get_header(std::string key)
+        {
+            return headers.find(key)->second;
+        }
```
Comment (The-EDev) [inline include/crow/http_response.h:54]: This is redundant, you can use `crow::get_header_value(headers, key)` (the `crow::` part isn't necessary in `unittest.cpp`)

Comment (hg333) [reply include/crow/http_response.h:54]: Okay, will remove it

**Comment by The-EDev** [comment]
Sorry, seems there's a problem with coveralls reporting. Nothing related to your PR, just gotta get it fixed.

**Review by The-EDev — APPROVED** [review]

**Comment by The-EDev** [comment]
Coveralls responded, seems like the issue was caused by some downtime they experienced over the weekend. Should be fine in time, I'll merge this now.

**Pull request merged**